package-testing/sdk-test-runner/src/app.ts

```diff
@@ -8,35 +8,89 @@ import { BanditActionRequest } from './dto/banditActionRequest';
 import { TestResponse } from './dto/testResponse';
 import { Scenario, Scenarios } from './dto/scenario';
 import { TestCase, TestSuite, TestSuiteReport, getJunitXml } from 'junit-xml';
+import {
+  ClientSDKRelay,
+  FeatureNotSupportedError,
+  SDKConnectionFailure,
+  SDKInfo,
+  SDKRelay,
+  ServerSDKRelay,
+} from './protocol';
+import { Config, SdkType } from './config';
 
 export default class App {
-  public constructor(
-    private readonly sdkName: string,
-    private readonly sdkServer: string,
-    private readonly apiServer: string,
-    private readonly testDataPath: string,
-    private readonly scenarioFile: string,
-    private readonly junitFile: string,
-  ) {}
+  private readonly config: Config;
 
-  private printHeader(): void {
-    log(`Firing up test runner for ${this.sdkName}`);
-    log(`Posting test cases to SDK server at ${this.sdkServer}`);
-    log(`Controlling configuration server at ${this.apiServer}`);
+  public constructor(config: Config) {
+    this.config = config;
   }
 
   public async run() {
     this.printHeader();
+    if (this.config.sdkType == SdkType.SERVER) {
+      return this.runServerTest();
+    } else {
+      return this.runClientTest();
+    }
+  }
+
+  public async runClientTest() {
+    log('Waiting for SDK Client to connect');
+
+    const sdkRelay = new ClientSDKRelay();
+    const sdkInfo = await sdkRelay.isReady();
+    if ('errorMessage' in sdkInfo && typeof sdkInfo.errorMessage === 'string') {
```

Comment:
I don't think you need this first check because otherwise `typeof sdkInfo.errorMessage` would be `'undefined'` and fail the second check.

If it's to appease TypeScript, you could use optional chaining.

```ts
if (typeof sdkInfo?.errorMessage === 'string') {
```

Reply:
thanks